### PR TITLE
[fundamental] Add workflow to weekly update runtime version

### PR DESCRIPTION
**Pull request by zhengfeiwang**
# Description

Add workflow to weekly update runtime version for test and example.

Have updated runtimes in canary region to adopt [latest runtime environment](https://ml.azure.com/environments/chat-with-pdf/version/8?wsid=/subscriptions/96aede12-2f73-41cb-b983-6d11a904839b/resourceGroups/promptflow/providers/Microsoft.MachineLearningServices/workspaces/promptflow-eastus2euap&tid=72f988bf-86f1-41af-91ab-2d7cd011db47).

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: .github/workflows/update-ci-runtime.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+          echo '${{ secrets.TEST_WORKSPACE_CONFIG_JSON_CANARY }}' > ${{ github.workspace }}/scripts/runtime_mgmt/canary.json
+          echo '${{ secrets.EXAMPLE_WORKSPACE_CONFIG_JSON_PROD }}' > ${{ github.workspace }}/scripts/runtime_mgmt/production.json
+
+      - name: Update runtime
+        shell: pwsh
+        working-directory: ${{ github.workspace }}/scripts/runtime_mgmt
+        run: |
+          pip install promptflow[azure]
+          python update-runtime.py --path canary.json
+          # python update-runtime.py --path production.json
```
Comment (zhengfeiwang) [inline .github/workflows/update-ci-runtime.yml:43]:
```suggestion
          python update-runtime.py --path production.json
```

Comment (zhengfeiwang) [reply .github/workflows/update-ci-runtime.yml:43]: mark - will commit suggestion tomorrow, after verify from the example CIs

File: scripts/runtime_mgmt/update-runtime.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    )
+
+
+def create_environment(ml_client: MLClient) -> str:
+    environment = load_environment(source=ENVIRONMENT_YAML)
+    env = ml_client.environments.create_or_update(environment)
+    # get workspace id from REST workspace object
+    resource_group_name = ml_client._operation_scope.resource_group_name
+    workspace_name = ml_client._operation_scope.workspace_name
+    location = ml_client.workspaces.get(name=workspace_name).location
```
Comment (wangchao1230) [inline scripts/runtime_mgmt/update-runtime.py:71]: small knowledge: you can avoid setting the name parameter here.

**Review by wangchao1230 — DISMISSED** [review]

File: scripts/runtime_mgmt/update-runtime.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    pfs_runtime_helper = PFSRuntimeHelper(ml_client=ml_client)
+
+    print("creating environment...")
+    env_asset_id = create_environment(ml_client=ml_client)
+    print("created environment, asset id:", env_asset_id)
+
+    print("updating runtime for test...")
+    pfs_runtime_helper.update_runtime(name=TEST_RUNTIME_NAME, env_asset_id=env_asset_id)
+    print("updating runtime for example...")
+    pfs_runtime_helper.update_runtime(name=EXAMPLE_RUNTIME_NAME, env_asset_id=env_asset_id)
```
Comment (0mza987) [inline scripts/runtime_mgmt/update-runtime.py:99]: what was the reason we need two runtimes with same environment? For log investigation?

Comment (zhengfeiwang) [reply scripts/runtime_mgmt/update-runtime.py:99]: one for test, the other for example; I don't want to mix them.

**Review by 0mza987 — DISMISSED** [review]

File: scripts/runtime_mgmt/update-runtime.py (hunk 2 above)
Comment (zhengfeiwang) [reply scripts/runtime_mgmt/update-runtime.py]: learnt that, interesting

**Review by 0mza987 — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**